Testing/VersionMinerTests/GitHubActionTests.cs

```diff
@@ -0,0 +1,506 @@
+        _mockConsoleService.VerifyOnce(m => m.Write("the repository exists.", true));
+
+        _mockConsoleService.VerifyOnce(m => m.Write($"✔️️Verifying if the repository branch '{inputs.BranchName}' exists . . . ", false));
+        _mockConsoleService.VerifyOnce(m => m.Write("the branch exists.", true));
+
+        _mockConsoleService.VerifyOnce(m => m.Write($"✔️️Getting data for file '{inputs.FilePath}' . . . ", false));
+        _mockConsoleService.VerifyOnce(m => m.Write("data retrieved", true));
+
+        _mockConsoleService.VerifyOnce(m => m.Write("✔️️Validating version keys . . . ", false));
+        _mockConsoleService.VerifyOnce(m => m.Write("version keys validated.", true));
+
+        _mockConsoleService.VerifyOnce(m => m.Write("✔️️Pulling version from file . . . ", false));
+        _mockConsoleService.VerifyOnce(m => m.Write("version pulled from file.", true));
+
+        _mockConsoleService.Verify(m => m.BlankLine(), Times.Exactly(9));
+    }
+
+    [Fact]
+    public async void Run_WhenNoVersionIsFoundForFailWhenVersionNotFoundInputAndIsSetToTrue_ThrowsException()
+    {
+        // Arrange
+        var expectedExceptionMsg = "No version value was found.";
+        expectedExceptionMsg += $"{Environment.NewLine}If you do not want the GitHub action to fail when no version is found,";
+        expectedExceptionMsg += "set the 'fail-when-version-not-found' input to a value of 'false'.";
+
+        _mockXMLParserService.Setup(m => m.KeyExists(It.IsAny<string>(), XMLVersionTagName, true)).Returns(true);
+        _mockXMLParserService.Setup(m => m.GetKeyValue(It.IsAny<string>(), XMLVersionTagName, true)).Returns(string.Empty);
+        var inputs = CreateInputs("Version",
+            failWhenVersionNotFound: true);
+        var action = CreateAction();
+
+        // Act
+        var act = () => action.Run(inputs, () => { }, e => throw e);
+
+        // Assert
+        await act.Should()
+            .ThrowAsync<NoVersionFoundException>()
+            .WithMessage(expectedExceptionMsg);
+    }
+
+    [Fact]
+    public async void Run_WhenAllKeysExistButAreEmptyForOnKeyValueMismatchInput_DoesNotFail()
+    {
+        // Arrange
+        _mockXMLParserService.Setup(m => m.KeyExists(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<bool>()))
+            .Returns(true);
+        _mockXMLParserService.Setup(m => m.GetKeyValue(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<bool>()))
+            .Returns(string.Empty);
+
+        var inputs = CreateInputs("Version,FileVersion,AssemblyVersion",
+            failOnKeyValueMismatch: true,
+            failWhenVersionNotFound: false);
+        var action = CreateAction();
+
+        // Act
+        var act = () => action.Run(inputs, () => { }, _ => { });
+
+        // Assert
+        await act.Should().NotThrowAsync("all requested versions that are empty still match.");
```

Comment:
This is correct.  This is a `because` class that actually adds this to other text that is constructed in the unit test failure message.  It is part of how the **FluentAssertions** unit testing library works.

Below is an example of a failed unit test message that this would produce.  It is intentional.
![image](https://user-images.githubusercontent.com/85414302/170895395-be23341f-6a3c-49e9-bd8a-864f17cfbf8c.png)